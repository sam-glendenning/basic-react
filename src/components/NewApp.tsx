/* eslint-disable no-mixed-spaces-and-tabs */
import React, { Component, ReactElement } from "react";

class NewApp extends Component {
  label = "Here's another heading!";

  handleClick = () : void => {
  	alert("Secret message, hehe");
  }

  public render() : ReactElement {
  	return (
    	<div className="NewApp">
    		<h2 onClick={this.handleClick}>{this.label}</h2>
    		<p id="final">Here is a final paragraph.</p>
    	</div>
  	);
  }
}

export default NewApp;
  