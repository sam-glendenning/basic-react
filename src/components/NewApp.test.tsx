import React from "react";
import renderer from "react-test-renderer";
import { shallow, ShallowWrapper } from "enzyme";
import NewApp from "./NewApp";
import { describe, it } from "@jest/globals";

describe("NewApp", () => {
	it("renders correctly", () => {
		const tree = renderer.create(<NewApp />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("renders correct text in a header", () => {
		const wrapper : ShallowWrapper = shallow(<NewApp />);
		expect(wrapper.find("h1").text()).toBe("Here's another heading!");
	});

	it("renders the first paragraph", () => {
		const wrapper : ShallowWrapper = shallow(<NewApp />);
		expect(wrapper.find("p").text()).toBe("Here is a final paragraph.");
	});

	it("should alert a secret message on header click", () => {
		jest.spyOn(window, "alert").mockImplementation(() => {return true;});
		const wrapper : ShallowWrapper = shallow(<NewApp />);
		const header = wrapper.find("h1");
		header.simulate("click");
		expect(window.alert).toBeCalledTimes(1);
	});
});
