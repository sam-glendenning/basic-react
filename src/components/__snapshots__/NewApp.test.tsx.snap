// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewApp renders correctly 1`] = `
<div
  className="NewApp"
>
  <h1
    onClick={[Function]}
  >
    Here's another heading!
  </h1>
  <p>
    Here is a final paragraph.
  </p>
</div>
`;
